hazelcast/src/test/java/com/hazelcast/test/starter/HazelcastStarterUtils.java

```diff
@@ -202,4 +249,22 @@ public static void assertInstanceOfByClassName(String className, Object object)
             throw new UnsupportedOperationException("Cannot locate collection type for " + type);
         }
     }
+
+    public static Map<Object, Object> newMapFor(Class<?> type) {
+        if (LinkedHashMap.class.isAssignableFrom(type)) {
+            return new LinkedHashMap<Object, Object>();
+        } else if (ConcurrentSkipListMap.class.isAssignableFrom(type)) {
+            return new ConcurrentSkipListMap<Object, Object>();
+        } else if (ConcurrentHashMap.class.isAssignableFrom(type)) {
+            return new ConcurrentHashMap<Object, Object>();
+        } else if (TreeMap.class.isAssignableFrom(type)) {
+            return new TreeMap<Object, Object>();
+        } else if (HashMap.class.isAssignableFrom(type)) {
+            return new HashMap<Object, Object>();
+        } else if (Map.class.isAssignableFrom(type)) {
+            return new ConcurrentHashMap<Object, Object>();
```

Comment:
I think I copied that one from the previous compatibility code, not sure what's the reason behind it but I guess it was just trying to be on the safe side - if we still haven't figured out which map the user is sending across classloaders, better make it a thread-safe map in case it's going to be used from different threads. Later I added the cases for specific implementations so this clause is probably going to be a lot less used.